.github/workflows/cmake.yml

```diff
@@ -62,3 +62,8 @@ jobs:
       # Execute tests defined by the CMake configuration.  
       # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
       run: ctest --output-on-failure
+
+    - name: Test library installation
+      working-directory: ${{github.workspace}}/build_linux
+      shell: bash
+      run: cmake --install . --prefix "$HOME/codec2_install" && rm -rf "$HOME/codec2_install"
```

Comment:
@tmiw - how will this trap the problem we encountered in https://github.com/drowe67/codec2/commit/0687cb9b6a7670c532db3b799a024ca20a01957e?

I'd rather not add more GitHub actions unless there is a strong reason, as they are executed on every push.  My understanding is the root cause here is relaying on an include file that was in the wrong path?  That's not a very common problem.

Reply:
I was reviewing again and realized that the error happened on `make install`--not simply while building. That was why I (and the current GH actions) weren't seeing any issues before. The new GH action will also allow us to validate that all the public include and library files get installed as expected in the future.

BTW the new GH action ran in 0 seconds per [the last results](https://github.com/drowe67/codec2/runs/7544022419?check_suite_focus=true).

Reply:
Cool - good work tracking it down.  The link between the cmake code and `make install` was pretty obscure :thinking: